proc/proc.go

```diff
@@ -109,48 +112,111 @@ func GetKernelModules(modulesPath string,
+	modules, err := parseKernelModules(content)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel modules: %v", err)
+	}
+
+	for _, kmod := range modules {
+		symmap.Add(libpf.Symbol{
+			Name:    libpf.SymbolName(kmod.name),
+			Address: libpf.SymbolValue(kmod.address),
+			Size:    kmod.size,
+		})
+	}
+
+	symmap.Finalize()
 
+	return &symmap, nil
+}
+
+func parseKernelModules(content []byte) ([]kernelModule, error) {
+	var (
+		modules                []kernelModule
+		scanner                = bufio.NewScanner(bytes.NewReader(content))
+		atLeastOneValidAddress = false
+		count                  = 0
+	)
+
+	for scanner.Scan() {
 		line := scanner.Text()
 
 		count++
 
-		nFields, err := fmt.Sscanf(line, "%s %d %d %s %s 0x%x",
-			&name, &size, &refcount, &dependencies, &state, &address)
+		kmod, err := parseKernelModuleLine(line)
 		if err != nil {
-			log.Warnf("err parsing line in modules: '%s'", err)
-			continue
+			return nil, fmt.Errorf("failed to parse kernel module line: %v", err)
 		}
-		if nFields < 6 {
-			log.Warnf("unexpected line in modules: '%s'", line)
-			continue
-		}
-		if address == 0 {
+		if kmod.address == 0 {
 			continue
 		}
 		atLeastOneValidAddress = true
 
-		symmap.Add(libpf.Symbol{
-			Name:    libpf.SymbolName(name),
-			Address: libpf.SymbolValue(address),
-			Size:    int(size),
-		})
+		modules = append(modules, kmod)
 	}
 
 	if count > 0 && !atLeastOneValidAddress {
 		return nil, errors.New("addresses from all modules is zero - check process permissions")
 	}
 
-	symmap.Finalize()
+	return modules, nil
+}
 
-	return &symmap, nil
+type kernelModule struct {
+	name    string
+	size    int
+	address uint64
+}
+
+func parseKernelModuleLine(line string) (kernelModule, error) {
+	// The format is: "name size refcount dependencies state address"
+	parts := strings.SplitN(line, " ", 6)
+	if len(parts) < 6 {
+		return kernelModule{}, fmt.Errorf("unexpected line in modules: '%s'", line)
+	}
+
+	size, err := parseSize(parts[1])
+	if err != nil {
+		return kernelModule{}, fmt.Errorf("failed to parse size value: '%s'", parts[1])
+	}
+
+	address, err := parseAddress(parts[5])
+	if err != nil {
+		return kernelModule{}, fmt.Errorf("failed to parse address value: '%s'", parts[5])
+	}
+
+	return kernelModule{
+		name:    parts[0],
+		size:    size,
+		address: address,
+	}, nil
+}
+
+func parseAddress(addressStr string) (uint64, error) {
+	address, err := strconv.ParseUint(strings.TrimPrefix(addressStr, "0x"), 16, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse address as hex value: '%s'", addressStr)
+	}
+
+	return address, nil
+}
+
+func parseSize(sizeStr string) (int, error) {
+	size, err := strconv.ParseInt(sizeStr, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse size int value: %q", sizeStr)
+	}
+
+	if size > int64(math.MinInt32) && size < int64(math.MaxInt32) {
+		return int(size), nil
+	}
+
+	return 0, fmt.Errorf("size value out of range: %q", sizeStr)
```

Comment:
What's the purpose of this check? If you want to enforce that the value is within a 32-bit range, why not call `parseInt` with `bitSize = 32` ?

Reply:
it was to please CodeQL which was complaining about not guarding this.